### ISSUE TITLE
Inconsistent subset output when operating on dense and sparse matrices

### ISSUE THREAD

**Issue by markovicpp** [josdejong/mathjs#2358]
Hi, I found this inconsistency when using subset to set values of an empty matrix.
Note: I'm not sure if this is an edge case (or supported at all) to use monotonically decreasing range.

Steps:
1.  Use subset to add an `identity` matrix to an empty `sparse` matrix, with one of the ranges _monotonically decreasing_ and one _monotonically increasing_.
2. Use the result to create a `dense` matrix (for debugging purpose).
3. Compare results with using `dense` matrix in step 1.

Results with empty `sparse` matrix
```
> math.matrix(math.subset(math.matrix('sparse'), math.index([1, 0], [0, 1]), math.identity(2)), 'dense')
Matrix {
  _data: [ [ 1, 0 ], [ 0, 1 ] ],
  _size: [ 2, 2 ],
  _datatype: undefined
}
```
Results with empty `dense` matrix
```
> math.matrix(math.subset(math.matrix('dense'), math.index([1, 0], [0, 1]), math.identity(2)), 'dense')
Matrix {
  _data: [ [ 0, 1 ], [ 1, 0 ] ],
  _size: [ 2, 2 ],
  _datatype: undefined
}
```

Node.js: 17.1.0
Math.js: 10.0.0

**Comment by josdejong** [comment]
Thanks for reporting Marko. I think this is a bug in the SparseMatrix implementation. 

Looking at the code it looks like it assumes that an index contains a consecutive range of numbers, and does not read the actual row/column numbers from `index`: it picks the lowest row/column index and then does do a for loop with the same number of items as in the index, but without looking at the actual numbers in the index.

https://github.com/josdejong/mathjs/blob/4ce71856afbcf1a0aaad30b70afc99c7606173c0/src/type/matrix/SparseMatrix.js#L394-L413

Anyone interested in looking into this bug? Help would be welcome!

**Comment by Al-0** [comment]
Hi, I will look into this bug!

**Comment by josdejong** [comment]
Thanks @Al-0 👍

**Comment by josdejong** [comment]
The fix of @Al-0 is published now in `v10.0.2`, see #2367.
